connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/model/JobWorkerAgentExecutionContext.java

```diff
@@ -90,4 +97,12 @@ public JobClient jobClient() {
   public ActivatedJob job() {
     return job;
   }
+
+  public boolean cancelRemainingInstances() {
+    return cancelRemainingInstances;
+  }
+
+  public void setCancelRemainingInstances(boolean cancelRemainingInstances) {
+    this.cancelRemainingInstances = cancelRemainingInstances;
+  }
```

Comment:
Not 100% happy with this one as it derives from the pattern of this being a read-only object, but it was the most obvious way to propagate this flag to the completion logic.